addons/hw_drivers/controllers/driver.py

```diff
@@ -79,7 +80,19 @@ def download_logs(self):
         """
         Downloads the log file
         """
-        if tools.config['logfile']:
-            return http.Stream.from_path(tools.config['logfile']).get_response(
+        log_path = tools.config['logfile']
+        if log_path:
+            check = adler32(log_path.encode())
+            stat = os.stat(log_path)
+            # intentionally don't use Stream.from_path as the path used is not in the addons path
+            # for instance, for the iot-box it will be in /var/log/odoo
+            return http.Stream(
```

Comment:
```suggestion
            return http.Stream(type='path',path=tools.config['logfile'],
```

Reply:
Why make it so complex?
Is there something I'm missing?

Reply:
Yes, the docstring of `http.Stream` says that you shouldn't instantiate `Stream` manually (using its `__init__` method) but instead use one of the 3 dedicated constructors `from_path`, `from_attachment`, or `from_binary_field`. Since the log file is nor an addon file, nor an attachment, nor a binary field, we *do* need to instanciate the object ourselves *and* make sure that all attributes are set.

The three constructors make it *easy* to stream files that users can genuinely access, and the framework itself sets all the quirks and bits to make it *fast* and *safe*. We want the API to be *hard* when it is about streaming anything else because it is unsafe.

Reply:
* `mimetype` let your browser know what software to use when you open the file, if you don't set it then your OS prompt you to select the software to use to open the file
* `size` let your browser shows a progress bar for the download, if you don't set it then it just shows that a download is in progress
* `last_modified`/`etag` stuff for the cache, if you don't set it you don't make good usage of the browser cache and you take the risk to download the *same* file again
* `download_name`: self explanatory

Reply:
Ok basically it is smarter